src/Service.Tests/Caching/CachingConfigDeserializationTests.cs

```diff
@@ -0,0 +1,242 @@
+    [DataRow(@",""cache"": { ""enabled"": true, ""ttl-seconds"": 60 }", true, DEFAULT_CACHE_TTL_SECONDS, true, DisplayName = "EntityCacheOptions provided: provided values used and userdefined flag set to true")]
+    [DataRow(@",""cache"": { ""enabled"": null }", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions.Enabled set to null results in (default) disabled entity cache.")]
+    [DataRow(@",""cache"": { ""enabled"": false, ""ttl-seconds"": null }", false, DEFAULT_CACHE_TTL_SECONDS, false, DisplayName = "EntityCacheOptions.TtlSeconds set to null results in (default) ttl value.")]
+    [DataTestMethod]
+    public void EntityCacheOptionsDeserialization_ValidJson(
+        string entityCacheConfig,
+        bool expectedEnabled,
+        int expectedTTL,
+        bool expectedUserDefinedTtl)
+    {
+        // Arrange
+        string fullConfig = GetRawConfigJson(globalCacheConfig: string.Empty, entityCacheConfig: entityCacheConfig);
+
+        // Act
+        RuntimeConfigLoader.TryParseConfig(
+            json: fullConfig,
+            out RuntimeConfig? config,
+            logger: null,
+            connectionString: null,
+            replaceEnvVar: false,
+            dataSourceName: string.Empty,
+            datasourceNameToConnectionString: null,
+            replacementFailureMode: EnvironmentVariableReplacementFailureMode.Throw);
+
+        // Assert
+        Assert.IsNotNull(config, message: "Config must not be null, runtime config JSON deserialization failed.");
+
+        Entity entity = config.Entities.First().Value;
+        Assert.AreEqual(expectedEnabled, entity.IsCachingEnabled, message: "EntityCacheConfig.Enabled expected to be: " + expectedEnabled);
+
+        EntityCacheOptions? resolvedEntityCacheOptions = entity.Cache;
+        if (expectedEnabled)
+        {
+            Assert.IsNotNull(resolvedEntityCacheOptions, message: "EntityCacheConfig must not be null, unexpected entity JSON deserialization result.");
+            Assert.AreEqual(expectedEnabled, resolvedEntityCacheOptions.Enabled, message: "EntityCacheConfig.Enabled expected to be: " + expectedEnabled);
+            Assert.AreEqual(expectedTTL, resolvedEntityCacheOptions.TtlSeconds);
+            Assert.AreEqual(expectedUserDefinedTtl, resolvedEntityCacheOptions.UserProvidedTtlOptions, message: "UserProvidedTtlOptions expected to be: " + expectedUserDefinedTtl);
+        }
+    }
+
+    /// <summary>
+    /// Validates that unexpected values provided for EntityCacheOptions in the JSON config result in
+    /// a failure to deserialize the runtime config.
+    /// </summary>
+    /// <param name="entityCacheConfig">Escaped JSON string defining entity cache configuration.</param>
+    [DataRow(@",""cache"": { ""enabled"": true, ""ttl-seconds"": 2147483648 }", DisplayName = "EntityCacheOptions.TtlSeconds set to Int.MaxValue+1 is invalid for parser.")]
```

Comment:
Can you add a case for `""ttl-seconds"": 0` and also some other small values, perhaps -1, and a small real number like .0001?

Reply:
thanks for the suggestion, adding. all three of those cases which would be invalid.